tests/s3_test.go

```diff
@@ -0,0 +1,85 @@
+package tests
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/aws/aws-sdk-go/service/s3/s3manager"
+	"github.com/gruntwork-io/terratest/modules/aws"
+	"github.com/gruntwork-io/terratest/modules/logger"
+	"github.com/gruntwork-io/terratest/modules/random"
+	"github.com/gruntwork-io/terratest/modules/retry"
+	"github.com/gruntwork-io/terratest/modules/terraform"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestTerraformS3Bucket(t *testing.T) {
```

Comment:
This function is going to need lots more comments because my team is going to have to learn go and terratest in order to maintain this.

Goal here is that someone not familiar with either can read through this code, understand it, and make modifications.

So for example, it'll be important to know what  `t.Parallel()` is doing and why it's necessary.

Reply:
Ok! Will make sure to be adding them here and in the other repos as well.
As for `t.Parallel()`, this tells Go that this function may run in parallel with others that would be calling `t.Paralllel()` in case we wanna run other functions/tests. (Useful for separating the unencrypted vs encrypted test for example.)